Closure_Front_End/src/huji-import/ParsedCoursesTables.vue

```diff
@@ -0,0 +1,101 @@
+<template> 
+    <unsupported-courses :courses="unsupportedCourses" />
+    <candidate-courses v-model:courses="candidateCourses"
+                       :totalCourseCount="parsedCourses.length"
+                       v-if="finishedProcessing"
+                       @import="$emit('import', $event)" />
+</template>
+
+<script>
+import { inject, toRefs, computed, onMounted, reactive, ref } from "vue";
+import UnsupportedCourses from './UnsupportedCourses.vue';
+import CandidateCourses from './CandidateCourses.vue'
+
+const processCourses = async (parsedCourses, http) => {
+    return await Promise.all(
+      // TODO: do this in a single API call
+      parsedCourses.value.map(async course => {
+        const res = await http.get(`/courses`, {
+          params: {
+            course_id: course.course_id,
+            // TODO: do not hard-core years, scrape Shnaton for several years
+            data_year: 2021 //course.year
+          }
+        });
+        if (res.data.count === 0) {
+            console.error(`Course ${course.course_id} - ${course.name} isn't in the DB`)
+            return {...course, notInDb: true}
+        }
+        if (res.data.count > 1) {
+            console.error(`Course ${course.course_id} - ${course.name} has more than 1 DB entry, how is that possible?`)
+            console.error(res.data.results)
+        }
+        const gottenCourse = res.data.results[0]
+        const finalCourse = { ... course, name: gottenCourse.name }
+        if (["FIRST", "SECOND"].includes(gottenCourse.semester)) {
+          if (course.semester && course.semester !== gottenCourse.semester) {
+            console.error(`Course ${course.course_id} - ${course.name} at year ${course.year} is offered only in semester `
+                         +`${gottenCourse.semester}, but student took it in ${course.semester}`)
+          } else {
+            finalCourse.semester = gottenCourse.semester
+          }
+        } else if (gottenCourse.semester === "EITHER") {
+          finalCourse.ambiguous = !course.semester
+        } else if (["ANNUAL", "SUMMER"].includes(gottenCourse.semester)) {
+          // TODO: support annual/summer courses
+          finalCourse.unsupportedPeriod = true
+          console.error(`Course ${course.course_id} - ${course.name} at year ${course.year} which is offered at unsupported period `
+                        +`${gottenCourse.semester}, and will be ignored`)
+        }
```

Comment:
Even HUJI isn't decisive about which semesters to register `annual` and `summer` semesters at, can we hardcode it to be an arbitrary semester (A or B)? 

It doesn't really matter for progress calculations

Reply:
I could hard-code it but shouldn't we instead implement a proper UI for summer/annual courses?

e.g, maybe for each year add a box below for summer courses, and annual courses would appear at both semesters(while only counting as one course)?

Reply:
Summer/annual boxes:
We didn't do them until now as there exists a very small number of such courses, and it seemed like too much of an effort, given the goal of getting something working to go live.

If we choose to add such boxes, my instinct says we shouldn't "prevent" users from dragging courses there, but maybe automatically move annual courses dragged to a non-annual box towards their correct position.

Reply:
Alright, I added a choice between merging and overriding(with a warning message in case overriding)